pistachio: marduk: Switch on heartbeat LED

francois-berder-imgtec opened the pull request:
this connects Creatordev/openwrt#45

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

francois-berder committed:
pistachio: marduk: Switch on heartbeat LED

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

Ham22 commented on board/imgtec/pistachio_bub/mfio.c:
```diff
@@ -308,3 +308,25 @@ void mfio_setup_usb_pwr(void)
+{
+	u32 reg, val;
+
+	/* make GPIO output */
```
Can you refactor to use a select_* function?

Avinash95 commented on board/imgtec/pistachio_bub/mfio.c:
```diff
@@ -103,16 +103,22 @@ static struct pistachio_mfio_reg pistachio_mfio_regs[] =
+{
+	u32 reg, val;
+	reg = PISTACHIO_GPIO + 0x204 + (((mfio) / 16) * 0x24);
+	val = 0x10000 | ((output) ? (1) : (0));
```
can we get rid of ternary operator by directly ORing output ?

francois-berder-imgtec replied:
I am leaving as it is to be consistent with the rest of the code.

francois-berder committed:
pistachio: Refactor pistachio_select_gpio

pistachio_select_gpio configures gpio as an input or an output.
Inputs are configured not to generate interrupts.

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

francois-berder committed:
pistachio: Add pistachio_set_gpio_output_state

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

Avinash95 commented on board/imgtec/pistachio_bub/mfio.c:
```diff
@@ -119,6 +119,16 @@ static void pistachio_select_gpio(u32 mfio, bool output)
+	u32 reg, val;
+
+	reg = PISTACHIO_GPIO + 0x208 + (((mfio) / 16) * 0x24);
+	val = 0x10000 | ((state) ? (1) : (0));
```
state is already bool, why do we need ternary operator ?

Avinash95 commented on board/imgtec/pistachio_bub/mfio.c:
```diff
@@ -103,23 +103,20 @@ static struct pistachio_mfio_reg pistachio_mfio_regs[] =
 };
 
-static void pistachio_select_gpio(u32 mfio, bool gpio)
+static void pistachio_select_gpio(u32 mfio, bool output)
```
instead of pistachio_select_gpio(), can we name it pistachio_configure_gpio() ?

francois-berder-imgtec replied:
done

Avinash95 commented on board/imgtec/pistachio_bub/mfio.c:
```diff
@@ -315,3 +315,18 @@ void mfio_setup_usb_pwr(void)
+void mfio_setup_led(void)
+{
+	/* make GPIO output */
+	pistachio_select_gpio(76, 1);
```
Can we have macro for this, as GPIO_OUTPUT and GPIO_INPUT

francois-berder committed:
pistachio: Rename pistachio_select_gpio to pistachio_configure_gpio

Signed-off-by: Francois Berder <francois.berder@imgtec.com>

Ham22 reviewed (approved):
Tested and see it working and nothing else seems to be broken
